File: app/src/main/java/es/uniovi/uo257977/clock/logic/TimerReceiver.java
package es.uniovi.uo257977.clock.logic;

import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import es.uniovi.uo257977.clock.Fragments.TimerFragment;

public class TimerReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {


        NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
        notificationManager.cancel(TimerFragment.TIMER_NOTIFICATION_ID);

        context.sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));//Cerramos la barra de notificaciones
    }
}

File: app/src/main/java/es/uniovi/uo257977/clock/Logic/TimerReceiver.kt
package es.uniovi.uo257977.clock.Logic

import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import es.uniovi.uo257977.clock.Fragments.TimerFragment

class TimerReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {

        val notificationManager = context?.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        notificationManager.cancel(TimerFragment.TIMER_NOTIFICATION_ID)
        context.sendBroadcast(Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS)) //Cerramos la barra de notificaciones

    }
}